Sources/CryptoSwift/LinuxFoundation.swift

```diff
@@ -0,0 +1,183 @@
+                if separator.count == 0 {separator = [13,10]}
+                
+                return (lineLength,separator)
+            }()
+            
+            var currentLineCount = 0
+            let appendByteToResult : (UInt8) -> () = {
+                result.append($0)
+                currentLineCount += 1
+                if let options = lineOptions where currentLineCount == options.lineLength {
+                    result.append(contentsOf: options.separator)
+                    currentLineCount = 0
+                }
+            }
+            
+            var currentByte : UInt8 = 0
+            
+            for (index,value) in bytes.enumerated() {
+                switch index%3 {
+                case 0:
+                    currentByte = (value >> 2)
+                    appendByteToResult(NSData.base64EncodeByte(currentByte))
+                    currentByte = ((value << 6) >> 2)
+                case 1:
+                    currentByte |= (value >> 4)
+                    appendByteToResult(NSData.base64EncodeByte(currentByte))
+                    currentByte = ((value << 4) >> 2)
+                case 2:
+                    currentByte |= (value >> 6)
+                    appendByteToResult(NSData.base64EncodeByte(currentByte))
+                    currentByte = ((value << 2) >> 2)
+                    appendByteToResult(NSData.base64EncodeByte(currentByte))
+                default:
+                    fatalError()
+                }
+            }
+            //add padding
+            switch bytes.count%3 {
+            case 0: break //no padding needed
+            case 1:
+                appendByteToResult(NSData.base64EncodeByte(currentByte))
+                appendByteToResult(base64Padding)
+                appendByteToResult(base64Padding)
+            case 2:
+                appendByteToResult(NSData.base64EncodeByte(currentByte))
+                appendByteToResult(base64Padding)
+            default:
+                fatalError()
+            }
+            return result
+        }
+        
+        // Create a Base-64 encoded NSString from the receiver's contents using the given options.
+        func base64EncodedString(_ options: NSDataBase64EncodingOptions = []) -> String {
+            
+            var decodedBytes = [UInt8](repeating: 0, count: self.length)
+            getBytes(&decodedBytes, length: decodedBytes.count)
+            let encodedBytes = NSData.base64EncodeBytes(decodedBytes, options: options)
+            let characters = encodedBytes.map { Character(UnicodeScalar($0)) }
+            return String(characters)
+        }
+    }
+    
+    public extension NSMutableData {
+        
+        func append(_ bytes: UnsafePointer<Void>, length: Int) {
+            
+            appendBytes(bytes, length: length)
+        }
+    }
+    
+    public extension NSDate {
+        
+        func timeIntervalSince(_ date: NSDate) -> NSTimeInterval {
```

Comment:
Yes, there are differences between in the way Swift 3.0 imports Cocoa APIs, and the Open Source version of Foundation that still has the Swift 2.2 syntax.